dev/how-to/maintain/tidb-in-kubernetes/log-collecting.md

```diff
@@ -0,0 +1,97 @@
+category: how-to
+---
+
+# Collect TiDB Logs in Kubernetes
+
+Running logs of the system and program can be very useful for troubleshooting problems and automating some operations. This document introduces the methods to collect logs of TiDB and its related components.
+
+## Collect logs of TiDB components in Kubernetes
+
+The TiDB components deployed by TiDB Operator output the logs in the `stdout` and `stderr` of the container by default. For Kubernetes, these logs are stored in the host's `/var/log/containers` directory, and the file name contains information such as the Pod name and the container name. For this reason, the log collection for the application in the container can be done directly on the host.
+
+If you already have a system for collecting logs in your existing infrastructure, you only need to add the `/var/log/containers/*.log` file on the host that holds Kubernetes in the collection scope by conventional means; if there is no available log collection system, or you want to deploy a separate system for collecting relevant logs, you are free to use any log collection system or solution that you are familiar with.
+
+[ElasticSearch](https://kubernetes.io/docs/tasks/debug-application-cluster/logging-elasticsearch-kibana/) and [Stackdriver](https://kubernetes.io/docs/tasks/debug-application-cluster/logging-stackdriver/) are provided as two log collection approaches for reference in the Kubernetes official documentation.
+
+Common open source tools that can be used to collect Kubernetes logs are:
+
+- [Fluentd](https://www.fluentd.org/)
+- [Fluent-bit](https://fluentbit.io/)
+- [Filebeat](https://www.elastic.co/products/beats/filebeat)
+- [Logstash](https://www.elastic.co/products/logstash)
+
+Collected Logs can usually be aggregated and stored on a specific server or stored in a dedicated storage and analysis system such as ElasticSearch.
+
+Some cloud service providers or specialized performance monitoring service providers also have their own free or chargeable log collection options that you can choose.
+
+If you do not aggregate logs via a separate log collection tool, you can also use the `kubectl` tool to directly view the running log of a container, but you cannot view the log of a destroyed container with this approach:
+
+{{< copyable "shell-regular" >}}
+
+```shell
+kubectl logs -n ${namespace} ${tidbPodName}
+```
+
+> **Note:**
+>
+> To view the log before the container restarts, you can add the `-p` parameter when executing the above command.
+
+If you need to collect logs from multiple Pods, you can use [`stern`](https://github.com/wercker/stern):
+
+{{< copyable "shell-regular" >}}
+
+```shell
+stern -n ${namespace} tidb -c slowlog
+```
+
+## Collect TiDB slow query logs
+
+For versions prior to 3.0, by default, TiDB prints slow query logs to standard output, mixed with application logs.
+
+- If the TiDB version <= 2.1.7, you can filter the slow query logs with the keyword `SLOW_QUERY`, for example:
```

Comment:
I think it's better to use "for tidb version <= 2.1.7" (and the same to following 2.1.8 part)